pinot-core/src/test/java/org/apache/pinot/core/operator/combine/CombineSlowOperatorsTest.java

```diff
@@ -194,17 +197,19 @@ private void testCancelCombineOperator(BaseCombineOperator combineOperator, Coun
     } finally {
       combineExecutor.shutdownNow();
     }
-    TestUtils.waitForCondition((aVoid) -> exp.get() instanceof QueryCancelledException, 10_000,
-        "Should have been cancelled");
+    TestUtils.waitForCondition((aVoid) -> exp.get() instanceof QueryCancelledException
+            || exp.get() instanceof EarlyTerminationException, 10_000, "Should have been cancelled");
     assertEquals(exp.get().getMessage(), errMsg);
```

Comment:
@Jackie-Jiang we need to catch both. previously early termination exception is unlikely to throw b/c we enter the base nextBlock() method first. then start the threading; now we start the threading first then calls nextBlock() thus we can potentially receive an EarlyTerminationExcpetion from the BaseOperator

is this desirable?